fs/readers.go

```diff
@@ -0,0 +1,135 @@
+package fs
+
+import (
+	"errors"
+	"io"
+)
+
+// A ChunkReader implements the io.Reader interface to only serve a chunk of the data
+// It will allow read until specific size of from the internal Reader hiding it's full size
+// Since ChunkReader will only Read data from internal Reader when requested it will
+// play nicely with the Account to reflect current speed
+type ChunkReader struct {
```

Comment:
Apart from the Len method, this is identical to [io.LimitedReader](https://golang.org/pkg/io/#LimitedReader)

If you need the Len method as well then you could embed an io.LimitedReader and add the Len method (since all its members are public)